src/couch_external_server.erl

```diff
@@ -40,47 +38,52 @@ execute(Pid, JsonReq) ->
 init([Name, Command]) ->
     couch_log:info("EXTERNAL: Starting process for: ~s", [Name]),
     couch_log:info("COMMAND: ~s", [Command]),
+    ok = config:subscribe_for_changes([{"couchdb", "os_process_timeout"}]),
     process_flag(trap_exit, true),
     Timeout = list_to_integer(config:get("couchdb", "os_process_timeout",
         "5000")),
     {ok, Pid} = couch_os_process:start_link(Command, [{timeout, Timeout}]),
-    ok = config:listen_for_changes(?MODULE, Pid),
-    {ok, {Name, Command, Pid}}.
+    {ok, {Name, Command, Pid, whereis(config_event)}}.
```

Comment:
> @davisp:  and if we get an unknown 'EXIT' message we just erlang:error on it.
We need config_event Pid to be able to distinguish among EXIT messages.

Rather then updating state I also considered keep Pid in the process dictionary. Let me know if you like it better.